### PR TITLE
DGIR-52: IP Range Restrictions Solr field & IP range restrictions filtering

### PR DESCRIPTION
**Pull request by Prashant-bd**

### CODE REVIEW

**Review by bibliophileaxe — CHANGES_REQUESTED** [review]

File: src/Plugin/search_api/processor/EmbargoIpRestriction.php [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    // Define the Embargo IP property.
+    $properties['embargo_ip'] = new ProcessorProperty([
+      'label' => $this->t('Embargo IP'),
```
[the remark this answers is not in the document]
Comment (Prashant-bd) [reply src/Plugin/search_api/processor/EmbargoIpRestriction.php]: DOne.

File: src/Plugin/search_api/processor/EmbargoIpRestriction.php [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      'type' => 'string',
+      'processor_id' => $this->getPluginId(),
+      'stored' => TRUE,
+      'indexed' => TRUE,
+      'multiValued' => TRUE,
+      'is_list' => TRUE,
+    ]);
+
+    // Define the exempt users property.
+    $properties['exempt_users'] = new ProcessorProperty([
```
[the remark this answers is not in the document]
Comment (Prashant-bd) [reply src/Plugin/search_api/processor/EmbargoIpRestriction.php]: Done.

File: src/Plugin/search_api/processor/EmbargoIpRestriction.php [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    // Define the Embargo IP property.
+    $properties['embargo_ip'] = new ProcessorProperty([
```
[the remark this answers is not in the document]
Comment (Prashant-bd) [reply src/Plugin/search_api/processor/EmbargoIpRestriction.php]: Done.

File: src/Plugin/search_api/processor/EmbargoIpRestriction.php [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+ *   id = "embargo_ip_restriction",
+ *   label = @Translation("Embargo IP Restriction"),
+ *   description = @Translation("Processor to filter results based on embargo and user IP."),
+ *   stages = {
+ *      "add_properties" = -10,
+ *      "preprocess_query" = 20,
+ *   }
+ * )
+ */
+class EmbargoIpRestriction extends ProcessorPluginBase implements ContainerFactoryPluginInterface {
```
[the remark this answers is not in the document]
Comment (Prashant-bd) [reply src/Plugin/search_api/processor/EmbargoIpRestriction.php]: Done

File: src/Plugin/search_api/processor/EmbargoIpRestriction.php [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    // Define the Embargo IP property.
+    $properties['embargo_ip'] = new ProcessorProperty([
+      'label' => $this->t('Embargo IP'),
+      'description' => $this->t('Stores the Embargo IPs for filtering.'),
```
[the remark this answers is not in the document]
Comment (Prashant-bd) [reply src/Plugin/search_api/processor/EmbargoIpRestriction.php]: Done

**Pull request closed without merging**